src/sage/repl/configuration.py

```diff
@@ -82,7 +82,8 @@ def colors(self):
             True
         """
         if not self._allow_ansi():
-            return 'NoColor'
+            from IPython import version_info
+            return 'nocolor' if version_info[0] >= 9 else 'NoColor'
```

Comment:
I'm not sure the conditional is necessary, I think 'nocolor' works ok for ipython 8. But it's ok anyway.